### PR TITLE
Update the provider to use latest libraries

### PR DESCRIPTION
**Pull request by nirarg**
This PR include:
* Upgrade libraries
* Code and tests changes to adapt the libraries upgrade
* Change kubevirtci (cluster-up) utility

### CODE REVIEW

File: kubevirt/schema/datavolume/data_volume.go [hunk 1]
```diff
@@ -3,16 +3,16 @@ package datavolume
...
 )
 
 func DataVolumeFields() map[string]*schema.Schema {
 	return map[string]*schema.Schema{
 		"metadata": k8s.NamespacedMetadataSchema("DataVolume", false),
-		"spec":     dataVolumeSpecSchema(),
+		"spec":     DataVolumeSpecSchema(),
 		"status":   dataVolumeStatusSchema(),
 	}
 }
```
Comment (bardielle) [inline kubevirt/schema/datavolume/data_volume.go:18]: I saw you added DataVolumeTemplatesSchema to be part of virtualmachine package - 
Does DataVolumeTemplatesSchema (line 20) still in use?

File: kubevirt/schema/virtualmachine/spec.go [hunk 2]
```diff
@@ -3,18 +3,17 @@ package virtualmachine
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 	"github.com/kubevirt/terraform-provider-kubevirt/kubevirt/schema/virtualmachineinstance"
-	kubevirtapiv1 "kubevirt.io/client-go/api/v1"
+	kubevirtapiv1 "kubevirt.io/api/core/v1"
 )
 
 func virtualMachineSpecFields() map[string]*schema.Schema {
 	return map[string]*schema.Schema{
 		// "running": &schema.Schema{
```
Comment (bardielle) [inline kubevirt/schema/virtualmachine/spec.go:14]: Why does the attribute `running` in a comment? can you remove it?

File: ci-tests/datavolume/datavolume_test.go [hunk 3]
```diff
@@ -50,18 +53,26 @@ var _ = AfterSuite(func() {
 })
 
 var _ = Describe("Data Volume Test", func() {
+	testName := "datavolume"
+	tfExecPath := "terraform"
+	It("init", func() {
```
Comment (bardielle) [inline ci-tests/datavolume/datavolume_test.go:58]: init -> terraform init

File: ci-tests/datavolume/datavolume_test.go [hunk 4]
```diff
@@ -50,18 +53,26 @@ var _ = AfterSuite(func() {
...
+		if err := terraform.Init(testDir, testName, tfExecPath); err != nil {
+			Fail(fmt.Sprintf("failed to init terraform (runDir: %s, testName: %s, terraform path: %s) , with error: %s", testDir, testName, tfExecPath, err))
+		}
+	})
 	It("create", func() {
```
Comment (bardielle) [inline ci-tests/datavolume/datavolume_test.go:63]: terraform create

File: ci-tests/datavolume/datavolume_test.go [hunk 5]
```diff
@@ -72,19 +83,21 @@ var _ = Describe("Data Volume Test", func() {
...
 			},
 		}
-		if _, err = exec.Apply(testDir, "datavolume", tfVarFiles); err != nil {
+		var extraOpts []tfexec.ApplyOption
+		if err = terraform.Apply(testDir, tfExecPath, tfVarFiles, extraOpts...); err != nil {
 			Fail(fmt.Sprintf("failed to update data volumes [%s, %s] in namespace %s, with error: %s", vars.DvFromHttpName, vars.DvFromPVCName, namespace, err))
 		}
 		validateDVs(vars)
 	})
 	It("delete", func() {
```
Comment (bardielle) [inline ci-tests/datavolume/datavolume_test.go:98]: terraform destroy

File: ci-tests/terraform/lineprinter/lineprinter.go [hunk 6]
```diff
@@ -24,12 +24,12 @@ type Print func(args ...interface{})
 // One use-case is connecting a subprocess's standard streams to a
 // logger:
 //
-//   linePrinter := &linePrinter{
-//     Print: &Trimmer{WrappedPrint: logrus.StandardLogger().Debug}.Print,
-//   }
-//   defer linePrinter.Close()
-//   cmd := exec.Command(...)
-//   cmd.Stdout = linePrinter
+//	linePrinter := &linePrinter{
```
Comment (bardielle) [inline ci-tests/terraform/lineprinter/lineprinter.go:27]: If those lines are in a comment - maybe they can be removed? 
also for lines 13-14

File: Makefile [hunk 7]
```diff
@@ -29,10 +29,10 @@ clean: $(GO)
 build: $(GO) test-fmt
 	$(GO) build
 
-install: build $(GO)
-	@echo "==> Installing plugin to $(DESTINATION_PREFIX)/$(shell $(GO) env GOOS)_$(shell $(GO) env GOARCH)"
-	@mkdir -p $(DESTINATION_PREFIX)/$(shell $(GO) env GOOS)_$(shell $(GO) env GOARCH)
-	@cp ./terraform-provider-kubevirt $(DESTINATION_PREFIX)/$(shell $(GO) env GOOS)_$(shell $(GO) env GOARCH)
+install-local: build $(GO)
```
Comment (bardielle) [inline Makefile:32]: maybe install-local-provider?

File: kubevirt/schema/datavolume/data_volume.go (hunk 1 above)
Comment (nirarg) [reply kubevirt/schema/datavolume/data_volume.go:16]: You right, its not used anymore, removed it

File: kubevirt/schema/virtualmachine/spec.go (hunk 2 above)
Comment (nirarg) [reply kubevirt/schema/virtualmachine/spec.go:14]: There was an issue adding the "running" with "run_strategy"
Should be fixed in this issue https://github.com/kubevirt/terraform-provider-kubevirt/issues/15

File: ci-tests/datavolume/datavolume_test.go (hunk 3 above)
Comment (nirarg) [reply ci-tests/datavolume/datavolume_test.go]: Changed

File: ci-tests/datavolume/datavolume_test.go (hunk 4 above)
Comment (nirarg) [reply ci-tests/datavolume/datavolume_test.go]: Changed to `terraform apply (create new)`

File: ci-tests/terraform/lineprinter/lineprinter.go (hunk 6 above)
Comment (nirarg) [reply ci-tests/terraform/lineprinter/lineprinter.go:27]: This is an example of usage as part of the function docs

File: Makefile (hunk 7 above)
Comment (nirarg) [reply Makefile]: Changed, thanks

**Review by bardielle — APPROVED** [review]

**Pull request merged**